pilot/pkg/model/service.go

```diff
@@ -609,19 +610,54 @@ func ParseServiceKey(s string) (hostname Hostname, ports PortList, labels Labels
 	return
 }
 
+// InboundPrefix is the cluster name prefix for inbound clusters
+const InboundPrefix = "in."
+
 // BuildSubsetKey generates a unique string referencing service instances for a given service name, a subset and a port.
 // The proxy queries Pilot with this key to obtain the list of instances in a subset.
+// Inbound cluster names: in._{portName}._{subsetName}.hostname
+// Outbound is the default direction for clusters so it is elided from the clustername
```

Comment:
Please don’t. It makes it very annoying to write and aggregate stats parsers. What’s the issue with adding an out._port._subset..? It’s still valid dns. It’s not like this is going to a dns srv client.